riak_test/yz_solr_upgrade_downgrade.erl

```diff
@@ -0,0 +1,459 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2016 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%%-------------------------------------------------------------------
+
+-module(yz_solr_upgrade_downgrade).
+
+-export([confirm/0]).
+
+-include("yokozuna.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+%% internal exports needed for callbacks
+-export([create_indexed_bucket_type/1]).
+
+-define(CLUSTER_SIZE, 2).
+-define(RING_SIZE, 8).
+-define(N_VAL, 2).
+
+-define(CONFIG, [
+    {riak_core, [
+        {ring_creation_size, 8},
+        {handoff_concurrency, 10},
+        {vnode_management_timer, 1000},
+        {default_bucket_props, [{n_val, ?N_VAL}]}
+    ]},
+    {yokozuna, [
+        {enabled, true},
+        {anti_entropy_tick, 500},
+        {anti_entropy_build_limit, {100, 1000}},
+        {anti_entropy_concurrency, 8}
+    ]}
+]).
+
+%% these indices is never upgraded
+%% - index-a1 is not upgraded and is not written to after upgrade
+%% - index-a1 is not upgraded and is written to after upgrade
+-define(INDEX_A1, <<"index-a1">>).
+-define(BUCKET_TYPE_A1, <<"bucket-type-a1">>).
+-define(BUCKET_A1, {?BUCKET_TYPE_A1, <<"bucket-name-a1">>}).
+-define(INDEX_A2, <<"index-a2">>).
+-define(BUCKET_TYPE_A2, <<"bucket-type-a2">>).
+-define(BUCKET_A2, {?BUCKET_TYPE_A2, <<"bucket-name-a2">>}).
+
+%% each of these indices are upgraded:
+%% - b1 is upgraded without any changes
+%% - b2 is upgraded but the data is deleted
+%% - b3 is upgraded but the data is backed up (for downgrade testing)
+-define(INDEX_B1, <<"index-b1">>).
+-define(BUCKET_TYPE_B1, <<"bucket-type-b1">>).
+-define(BUCKET_B1, {?BUCKET_TYPE_B1, <<"bucket-name-b1">>}).
+-define(INDEX_B2, <<"index-b2">>).
+-define(BUCKET_TYPE_B2, <<"bucket-type-b2">>).
+-define(BUCKET_B2, {?BUCKET_TYPE_B2, <<"bucket-name-b2">>}).
+-define(INDEX_B3, <<"index-b3">>).
+-define(BUCKET_TYPE_B3, <<"bucket-type-b3">>).
+-define(BUCKET_B3, {?BUCKET_TYPE_B3, <<"bucket-name-b3">>}).
+
+%% index-c is created after the upgrade
+-define(INDEX_C, <<"index-c">>).
+-define(BUCKET_TYPE_C, <<"bucket-type-c">>).
+-define(BUCKET_C, {?BUCKET_TYPE_C, <<"bucket-name-c">>}).
+
+-define(LUCENE_MATCH_VERSION_OPEN, "<luceneMatchVersion>").
```

Comment:
LUCENE_MATCH_VERSION_(OPEN|CLOSED) don't appear to be used anymore. Fred mentioned to me in Friday that these are indeed leftover from a previous version of the code, and can be removed.